doc/user/content/manage/troubleshooting.md

```diff
@@ -183,7 +183,66 @@ ORDER BY elapsed_ns DESC
 From the results of this query we can see that most of the elapsed time of the dataflow is caused by the time it takes to maintain the updates in one of the arrangements of the dataflow of the materialized view.
 
 ## Why is Materialize unresponsive or slow?
+Slow responses can be the result of a number of factors, which we can help you understand and address.
 
+### Indexes
+
+If you haven’t already, consider creating an [index](https://materialize.com/docs/sql/create-index/) for your query. The [optimizations](https://materialize.com/docs/transform-data/optimization/) page in our docs has more advanced guidance on building an efficient index. Indexes maintain a query’s up-to-date results in memory within a [cluster](https://materialize.com/docs/get-started/key-concepts#clusters), which allows future queries to return quickly.
+
+If you’re planning to run a query regularly and expect it to be fast, an index is highly recommended.
+
+Some tips for troubleshooting using an index:
+
+- Start by querying the [EXPLAIN](https://materialize.com/docs/sql/explain/) plan. This will tell you the query plan that Materialize will execute for your query.
+- Once you add an index, re-issue the explain plan to ensure the index was actually used.
+- If you have already created an index, and it’s not showing up in the explain plan, you may be running the query in a different cluster than where the index resides. An index caches the results in memory on the cluster on which you create it. You need to execute your query on that same cluster in order to utilize the index.
```

Comment:
This could also be because the index columns don't exactly match a predicate. Many people will come in expecting prefix matches to work but they don't with us. Correct as is, but this could also be a common reason.